src/web/components/ParticipantManagement/AddParticipantDialog.tsx

```diff
@@ -0,0 +1,239 @@
+    [sites]
+  );
+  const [open, setOpen] = useState(false);
+  const [siteSearchResults, setSiteSearchResults] = useState<Fuse.FuseResult<SiteDTO>[]>();
+  const [searchText, setSearchText] = useState('');
+  const [selectedSite, setSelectedSite] = useState<SiteDTO>();
+
+  const formMethods = useForm<AddParticipantForm>();
+  const { register, setValue, watch, handleSubmit, reset } = formMethods;
+
+  useEffect(() => {
+    setSiteSearchResults(fuse?.search(searchText));
+  }, [fuse, searchText]);
+
+  useEffect(() => {
+    const subscription = watch((value, { name }) => {
+      if (name === 'participantTypes') {
+        const types = value.participantTypes;
+        const roles = GetRecommendedRolesById(types as number[]);
+        setValue('apiRoles', roles);
+      }
+    });
+    return () => subscription.unsubscribe();
+  }, [watch, setValue, open]);
+
+  useEffect(() => {
+    if (!open) {
+      reset();
+    }
+  }, [open, reset]);
+
+  const onSubmit = async (formData: AddParticipantForm) => {
+    await onAddParticipant(formData);
+    setOpen(false);
+  };
+
+  const onSiteClick = (site: SiteDTO) => {
+    setValue('siteId', site.id);
+    setSelectedSite(site);
+  };
+
+  const onSearchInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchText(event.target.value);
+  };
+
+  return (
+    <Dialog
+      triggerButton={triggerButton}
+      title='Add Participant'
+      closeButtonText='Cancel'
+      open={open}
+      onOpenChange={setOpen}
+      className='add-participant-dialog'
+    >
+      <FormProvider {...formMethods}>
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <h4>Participant Information</h4>
+          <span>Provide the following information about the company/participant</span>
+          <div className='add-participant-dialog-flex'>
+            <div>
+              <TextInput
+                inputName='participantName'
+                label='Participant Name'
+                className='text-input'
+                rules={{ required: 'Please specify Participant Name.' }}
+              />
+            </div>
+
+            <div className='right-column'>
+              <CheckboxInput
+                inputName='participantTypes'
+                label='Participant Type'
+                options={participantTypes.map((p) => ({
+                  optionLabel: p.typeName,
+                  value: p.id,
+                }))}
+                rules={{ required: 'Please specify Participant Type(s).' }}
+              />
+            </div>
+          </div>
+          <div>
+            <div className='add-participant-dialog-flex'>
+              <div className='left-column site-type'>
+                <RadioInput
+                  inputName='siteIdType'
+                  label='Site ID'
+                  options={[
+                    { optionLabel: 'Existing Site ID', value: 0, checked: true },
+                    { optionLabel: 'New Site ID', value: 1, disabled: true },
```

Comment:
I presume we are just disabling this option during development?

Reply:
Correct.